core-v1/src/main/scala/algebra/Expr.scala

```diff
@@ -195,43 +215,43 @@ object Expr {
     ): I ~:> F[Boolean]
   }
 
-  // TODO: Use ExtractValue.AsBoolean instead of Boolean here
   // TODO: Use a NonEmptyList of expressions?
   /**
     * Evaluates the left and right expression nodes and uses `&&` to combine the results.
     *
     * @param leftExpr the left side of the `AND` operation
     * @param rightExpr the right side of the `AND` operation
     */
-  final case class And[-I, OP[_]](
-    leftExpr: Expr[I, Boolean, OP],
-    rightExpr: Expr[I, Boolean, OP],
+  final case class And[-I, +B, F[+_], OP[_]](
+    leftExpr: Expr[I, F[B], OP],
+    rightExpr: Expr[I, F[B], OP],
```

Comment:
I should probably make this support 2 or more expressions. We use that in a lot of places.